### PR TITLE
Move Widget Inspector service extensions from DevTools to Flutter

### PR DESCRIPTION
**Pull request by kenzieschmoll**
These service extensions have been dynamically registered from DevTools for years now (see [inspector_polyfill_script.dart](https://github.com/flutter/devtools/blob/master/packages/devtools_app/assets/scripts/inspector_polyfill_script.dart)). Originally, when DevTools had to be backwards compatible with older versions of Flutter, this was a way to guarantee these service extensions would be available for any version of Flutter. Since this time, we have changed the DevTools distribution strategy to ship DevTools with the Dart SDK, so we no longer have to be concerned with backwards compatibility issues.

Moving these service extension registrations to the widget_inspector.dart file in Flutter is a tech debt clean up, and also a performance improvement for DevTools IPL. This change will eliminate the need for the DevTools inspector to perform an expensive 'eval' at the initial load of the inspector to register the service extensions (@annagrin). The initial expression compilation that draws the Flutter layout explorer takes ~3sec when connected to a flutter web app built with DDC, which, before this change, delayed the time until the flutter inspector was ready for user action. This change prevents the 3 seconds delay from happening before the inspector loads. However, the delay will still exist for the first load of the layout explorer (the cost is loading the widget_inspector.dart file for evaluation). This is a separate problem that will be tackled in a separate effort. 

All the new service extensions added in this PR are copied from https://github.com/flutter/devtools/blob/master/packages/devtools_app/assets/scripts/inspector_polyfill_script.dart with minimal changes for adhering to the flutter/flutter style guide and lints. 

This is the required flutter changes for https://github.com/flutter/devtools/issues/4634. DevTools changes are in https://github.com/flutter/devtools/pull/4633.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -1127,6 +1123,10 @@ mixin WidgetInspectorService {
         return null;
       },
     );
+    registerServiceExtension(
```
Comment (kenzieschmoll) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:1126]: The service extensions added in this file are copied / moved from https://github.com/flutter/devtools/blob/master/packages/devtools_app/assets/scripts/inspector_polyfill_script.dart.

File: packages/flutter/test/widgets/widget_inspector_test.dart [hunk 2]
```diff
@@ -2216,6 +2223,97 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
       expect(childJson['chidlren'], isNull);
     }, skip: !WidgetInspectorService.instance.isWidgetCreationTracked()); // [intended] Test requires --track-widget-creation flag.
 
+    testWidgets('ext.flutter.inspector.getRootWidgetSummaryTreeWithPreviews', (WidgetTester tester) async {
```
Comment (kenzieschmoll) [inline packages/flutter/test/widgets/widget_inspector_test.dart:2226]: The tests added to this file are all new, as we did not have tests for these service extensions in DevTools.

File: packages/flutter/test/widgets/widget_inspector_test.dart [hunk 3]
```diff
@@ -2216,6 +2223,97 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
...
+      ))! as List<Object?>;
+      expect(alternateChildrenJson.length, equals(1));
+      Map<String, Object?> childJson = childrenJson[0]! as Map<String, Object?>;
+      Map<String, Object?> alternateChildJson = alternateChildrenJson[0]! as Map<String, Object?>;
+      expect(childJson['description'], startsWith('Directionality'));
+      expect(alternateChildJson['description'], startsWith('Directionality'));
+      expect(alternateChildJson['valueId'], equals(childJson['valueId']));
+
+      childrenJson = childJson['children']! as List<Object?>;
+      alternateChildrenJson = (await service.testExtension(
```
Comment (CoderDake) [inline packages/flutter/test/widgets/widget_inspector_test.dart:2288]: This is a very long test that seems to test seperate testExtension calls.
Would it be possible to split this up into multiple tests to help make debugging easier if these ever end up failing?

Comment (kenzieschmoll) [reply packages/flutter/test/widgets/widget_inspector_test.dart:2288]: I copied this test from 'ext.flutter.inspector.getRootWidgetSummaryTree' above, and stripped out what I could. We need to call 'getChildrenSummaryTree' to access the summary tree for the children. Unfortunately the child with the 'textPreview' occurs a few levels down in the tree. I could try removing the Stack widget that is being pumped, but that would only eliminate one level of the tree.

File: packages/flutter/lib/src/widgets/service_extensions.dart [hunk 4]
```diff
@@ -377,6 +397,21 @@ enum WidgetInspectorServiceExtensions {
   ///   extension is registered.
   getRootWidgetSummaryTree,
 
+  /// Name of service extension that, when called, will return the
+  /// [DiagnosticsNode] data for the root [Element] of the summary tree, which
+  /// only includes [Element]s that were created by user code, and includes a
+  /// text preview for each element.
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/flutter/lib/src/widgets/service_extensions.dart]: fixed

File: packages/flutter/lib/src/widgets/service_extensions.dart [hunk 5]
```diff
@@ -237,6 +237,8 @@ enum WidgetInspectorServiceExtensions {
   ///   service extension calls.
   /// * [WidgetInspectorService.removePubRootDirectories], which should be used
   ///   to remove directories.
+  /// * [WidgetInspectorService._getPubRootDirectories], which should be used
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/service_extensions.dart:240]: (here and elsewhere): Docs shouldn't reference private members.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by CoderDake — APPROVED** [review]

**Pull request merged**